unicodetools/src/main/java/org/unicode/props/PropertyParsingInfo.java

```diff
@@ -48,12 +49,34 @@ enum SpecialProperty {
     public final UcdProperty property;
     public final SpecialProperty special;
 
+    public static class FieldMapping implements Comparable<FieldMapping> {
+        FieldMapping(int valueField) {
+            this(0, valueField);
+        }
+
+        FieldMapping(int keyField, int valueField) {
+            this.keyField = keyField;
+            this.valueField = valueField;
+        }
+
+        @Override
+        public int compareTo(FieldMapping other) {
+            return Comparator.<FieldMapping>comparingInt(m -> m.keyField)
+                    .thenComparing(m -> m.valueField)
+                    .compare(this, other);
+        }
+
+        final int keyField;
+        final int valueField;
+    }
+    ;
```

Comment:
?

Reply:
Spot the C++ programmer (and then spotless kicked the semicolon down).